src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/MQConnection.java

```diff
@@ -62,11 +65,35 @@
     private String serverUri;
     private String virtualHost;
     private Connection connection = null;
-    private Channel channel = null;
 
-    private volatile LinkedBlockingQueue messageQueue;
+    private volatile LinkedBlockingQueue messageQueue = new LinkedBlockingQueue(100000);
+    private volatile ConcurrentNavigableMap<Long, MessageData> outstandingConfirms = new ConcurrentSkipListMap<>();
     private Thread messageQueueThread;
 
+
+    /**
+     * Throw on exceptions when creating a channel
+     */
+    private class ChannelCreationException extends Exception {
```

Comment:
Thinking about it, this should probably inherit from IOException.

Reply:
Good suggestion